### DISCUSSION TITLE
`manuscript-cli` vs. `manuscript-gui`

### DISCUSSION THREAD

**Discussion by KagemniKarimu** [chainbase-labs/manuscript-core#78] [General]
It is very exciting to see how manuscripts are shaping up. :eyes: :fire: I am increasingly grateful for our collaborations :).
There are a couple of points of discussion I wanted to raise concerning `manuscript-core`:

1. **Naming** :name_badge: 

Doesn't it make more sense for us to call the rust binary `manuscript-gui` ? Then, it would have parity with `manuscript-cli` and follow a common naming convention. Because it's not a desktop application and is still called from the terminal, calling it `manuscript` seems a bit confusing for developers. Especially since, as a GUI terminal application, it isn't really intended to take any args.

2. **Repo Organization** :file_folder: 

Is it feasible for us to migrate all the CLI and go related functions to a folder called `cli`?  We could then leave the top-level directory for stuff that is high-level `manuscript-core` and/or deals with the monorepo itself (things like `README`, `contributing.md`, git files, licenses, and install scripts). 

I know it's a bit of work, but it would establish parity with how the `gui` functionality is currently relegated to a single folder. It would also be more immediately obvious to devs where to find relevant files and which application they're working on. :telescope: 

3. **Shared Resources** :recycle: 

Just recently when version was bumped, it bumped to **_1.1.0_** for the GUI, but remained the same for the CLI.  This is because they were hard-coded in separate files in their respective project directories.  Since there are no divergent versions as we're using a monorepository, there should really only be one version number. What do we think about creating a config or build file which stores that information in a single place?

There might be other useful information that both projects can pull from?

4. **Install Scripts** :arrow_down: 

This is a long-range consideration, but something I think could be a very nice-to-have. 
Instead of (or in addition to) having two separate scripts, I think it might be better DX to have one script which allows the developer to multiple select via interactive install. That is to say, a developer would run the script and from the script see something like the following:
```
{{Beautiful ASCII Art}}
Welcome to **Chainbase Manuscript Installer** 
Which Manuscript Components would you like installed?
- GUI (recommended)
- CLI
- Local Documentation
```

Then if the developer didn't want to choose via a prompt, they could bypass interactive install by passing flags. I imagine something like `--install=cli`  might be passed to the script - where the default option installs both `manuscript-gui` and `manuscript-cli`.

**Comment by Liquidwe** [comment]
@KagemniKarimu  Thank you for the feedback! These are all very valuable suggestions. I think I can submit a few issues to track the progress. Developers who are interested are welcome to take on these tasks and contribute together:

1.Standardize the name of "manuscript gui" to "manuscript-gui" 
2.Organize the CLI code into a separate project directory.
3.Share a unified version number.
4.Update the installation script to allow users to choose between GUI and CLI.

**Reply by KagemniKarimu** [reply]
Done!  :heavy_check_mark: 
See #100 , #99 , #98 .

Concern about the name standardization is technically already handled by our Makefile #83 :)
